src/core/Database.h

```diff
@@ -57,40 +60,33 @@ class Database : public QObject
     };
     static const quint32 CompressionAlgorithmMax = CompressionGZip;
 
-    struct DatabaseData
-    {
-        QUuid cipher;
-        CompressionAlgorithm compressionAlgo;
-        QByteArray transformedMasterKey;
-        QSharedPointer<Kdf> kdf;
-        QSharedPointer<const CompositeKey> key;
-        bool hasKey;
-        QByteArray masterSeed;
-        QByteArray challengeResponseKey;
-        QVariantMap publicCustomData;
-    };
-
     Database();
+    explicit Database(const QString& filePath);
     ~Database() override;
+
+    QUuid uuid() const;
+
+    bool open(QSharedPointer<const CompositeKey> key, bool readOnly = false, QString* error = nullptr);
+    bool open(const QString& filePath, QSharedPointer<const CompositeKey> key, bool readOnly = false, QString* error = nullptr);
+    bool save(bool atomic = true, bool backup = false, QString* error = nullptr);
+    bool save(const QString& filePath, bool atomic = true, bool backup = false, QString* error = nullptr);
+
+    QString filePath() const;
+    void setFilePath(const QString& filePath);
```

Comment:
open and save set the file path and therefore use this method. The reason why it's there is because we have a getter for it and I like symmetry. You may also alter the file path, but not quite save it yet. I won't fight for it, though. We can make this method private if you prefer that.

Reply:
We can leave this as is, there is no adverse effects of using any of these functions in any order.